src/cursor/cursor.ts

```diff
@@ -54,6 +54,81 @@ export interface CursorOptions extends CoreCursorOptions {
   numberOfRetries?: number;
 }
 
+const kCursor = Symbol('cursor');
+
+export class CursorStream extends Readable {
+  [kCursor]: Cursor;
+
+  /** @event */
+  static readonly CLOSE = 'close' as const;
+  /** @event */
+  static readonly DATA = 'data' as const;
+  /** @event */
+  static readonly END = 'end' as const;
+  /** @event */
+  static readonly FINISH = 'finish' as const;
+  /** @event */
+  static readonly ERROR = 'error' as const;
+  /** @event */
+  static readonly PAUSE = 'pause' as const;
+  /** @event */
+  static readonly READABLE = 'readable' as const;
+  /** @event */
+  static readonly RESUME = 'resume' as const;
+
+  constructor(cursor: Cursor) {
+    super({ objectMode: true });
+    this[kCursor] = cursor;
+  }
+
+  destroy(err?: AnyError): void {
+    if (err) this.emit(Cursor.ERROR, err);
+    this.pause();
+    this[kCursor].close();
+  }
+
+  /** @internal */
+  _read(): void {
+    const cursor = this[kCursor];
+    if ((cursor.s && cursor.s.state === CursorState.CLOSED) || cursor.isDead()) {
+      this.push(null);
+      return;
+    }
+
+    // Get the next item
+    cursor._next((err, result) => {
+      if (err) {
+        if (this.listeners(CursorStream.ERROR) && this.listeners(CursorStream.ERROR).length > 0) {
+          this.emit(CursorStream.ERROR, err);
+        }
+        const done = () => this.emit(CursorStream.END);
+        cursor.isDead() ? done() : cursor.close(done);
+        return;
+      }
+
+      // If we provided a transformation method
+      if (
+        cursor.streamOptions &&
```

Comment:
shouldn't the `CursorStream` be configured with `CursorStreamOptions`? Why check for those options back on the cursor itself?